### ISSUE TITLE
Provide some cloud functions/modules

### ISSUE THREAD

**Issue by Shinigami92** [faker-js/faker#953]
### Clear and concise description of the problem

Stuff like `faker.aws.region()`

### Suggested solution

Draft out what is needed

**Comment by Minozzzi** [comment]
Can u explain better? 
How this should work...

**Comment by Shinigami92** [comment]
> Can u explain better? How this should work...

Just like `faker.aws.region() // 'eu-west-1'`
See https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html#Concepts.RegionsAndAvailabilityZones.Regions for more regions

**Comment by Minozzzi** [comment]
Hey guys, I would like to try to contribute with this feature. Could you assign to me?

**Comment by import-brain** [comment]
> Hey guys, I would like to try to contribute with this feature. Could you assign to me?

Not too sure if this feature request is accepted yet, it's been here for a while. @Shinigami92?

**Comment by Shinigami92** [comment]
The more important thing here is currently to collect what is needed and how we should structure it
So you might not instantly open a PR for that but provide some I don't know collection of draft data and such :)

**Comment by Shinigami92** [comment]
Today we decided in our team that we wont like to start with an implementation of this issue until we
1. fully draft out what we want here (that's now your part @Minozzzi :slightly_smiling_face:)
2. fully support a much better tree-shaking that also strips out locales and module functions
   this could take like half a year :warning:

Please write a draft that detail how you would design the cloud module(s) in this issue, please do not write a PR yet
e.g. `faker.cloud.aws.region()` vs `faker.aws.region()` vs `faker.cloud.region(provider: 'aws')`

**Comment by Minozzzi** [comment]
> Today we decided in our team that we wont like to start with an implementation of this issue until we
> 
> 1. fully draft out what we want here (that's now your part @Minozzzi slightly_smiling_face)
> 2. fully support a much better tree-shaking that also strips out locales and module functions
>    this could take like half a year warning
> 
> Please write a draft that detail how you would design the cloud module(s) in this issue, please do not write a PR yet e.g. `faker.cloud.aws.region()` vs `faker.aws.region()` vs `faker.cloud.region(provider: 'aws')`

For the first point, I imagine we could use locals to do this, with `faker.cloud.region(provider: 'aws')`. I saw that regions can also be the same for different providers, for that we could use a base locale and create specific locales when necessary.

For the second point, I can't speak very well about it, I'm not a very experienced developer yet, so I wouldn't know how to do it

**Comment by domingo2000** [comment]
Giving that different cloud providers could have different region names and avaliabilities, it would i think it woul be nicer if the module `cloud` got separated for different providers, so i should say that the formar `faker.cloud.aws.region()` is more Open Closed than `faker.cloud.region(provider= 'aws')`, for example we could define more cloud resources with the first choice like
For AWS:
   - `faker.cloud.aws.region() -> us-east-1 `
   - `faker.cloud.aws.avaliabilityZone() -> us-east-1a `

If later is needed to define more resources for each cloud provider we could nest the resources in each provider module like

`faker.cloud.aws.service() -> lambda`

If is needed to add another provider to the module:
- `faker.cloud.{provider}.{providerResource}() -> fakeData`

To go deeper in each cloud resource own logic it could be also implemented
- `faker.cloud.{provider}.{providerResource}.{resourceDetail}() -> fake data`

Note: The information needed for AWS regions and Availability Zones can be found using the aws CLI
- regions: https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-regions.html
![image](https://user-images.githubusercontent.com/40220174/197277647-8dda8974-5233-481a-aa0c-6ce0f0088273.png)
- availability zones: https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-availability-zones.html
![image](https://user-images.githubusercontent.com/40220174/197277980-2d9c5064-a92c-4424-bfd8-0e7b3d6ad10b.png)

**Comment by Shinigami92** [comment]
The only issue about that is that we want to get rid of sub-modules :thinking: 
We already want to redesign the image-providers which is a bit similar to what you proposing.

**Comment by domingo2000** [comment]
True, I belive that some of the problems of the image-provider could appear here, because some cloud services change often. Some cloud Iaac solutions still separate by cloud providers, see [Terraform Namespaces](https://registry.terraform.io/namespaces/hashicorp), it has one namespace for each big cloud provider, because the buisness logic of each is very different. There might be some organization in the cloud module like the one that they have 😄

**Comment by ST-DDT** [comment]
> The only issue about that is that we want to get rid of sub-modules 🤔 We already want to redesign the image-providers which is a bit similar to what you proposing.

I dont think we should get rid of the sub modules per se.
The image provider is just inconsistent on how its implemented.

If were are going for submodules, then we have to adjust the docs generation to support it. I think we shouldnt go for sub sub modules though.

**Comment by ST-DDT** [comment]
See also:
* https://github.com/faker-js/faker/issues/1475
